www/js/search.js

```diff
@@ -537,21 +537,75 @@ module.exports = {
     if (searchQuery.length >= 1) {
       banidb
         .query(searchQuery, searchType, this.searchSource)
-        .then(rows => this.printResults(rows));
+        .then(rows => this.printResults(rows, searchType, searchQuery));
     } else {
       this.$results.innerHTML = '';
     }
   },
 
   akhandPaatt,
 
-  printResults(rows) {
+  /* Taken from https://github.com/KhalisFoundation/sttm-web/blob/dev/src/js/util/index.js
+   * Line no. 271
+   */
+
+  getHighlightIndices(baani, query, type) {
+    let start = -1;
+    let end = -1;
+
+    if (baani === null) {
+      return [start, end];
+    }
+
+    let baaniWords = baani.split(' ');
+
+    switch (type) {
+      case CONSTS.SEARCH_TYPES.FIRST_LETTERS: // eslint-disable-line no-fallthrough
+      case CONSTS.SEARCH_TYPES.FIRST_LETTERS_ANYWHERE:
+      default: {
+        // remove i from start of words
+        baaniWords = baaniWords.map(w => (w.startsWith('i') ? w.slice(1) : w));
+
+        start = baaniWords
+          .map(word => word[0])
+          .join('')
+          .indexOf(query);
+        end = start + query.length;
+        break;
+      }
+      case CONSTS.SEARCH_TYPES.GURMUKHI_WORD: {
+        const q = query.split(' ');
+        start = baaniWords.indexOf(q[0]);
+        start = start === -1 ? baaniWords.findIndex(w => w.includes(q[0])) : start;
+        end = start + q.length;
+        break;
+      }
+    }
+
+    return [start, start === -1 ? 0 : end];
+  },
+
+  printResults(rows, searchType, searchQuery) {
```

Comment:
![image](https://user-images.githubusercontent.com/2982784/73464040-20265e80-4333-11ea-980a-de9296d5e5ba.png)
undefined is getting populated for some results in Full word search when I add a space.